R/utils-assertions.R

```diff
@@ -1,16 +1,22 @@
-.pipes <- c("%>%")
+.pipes <- c("%>%", "|>")
 
 is_pipe <- function(x) {
   if (is.character(x)) x <- parse(text = x)[[1]]
   ifelse(is.call(x), as.character(x[[1]]) %in% .pipes, FALSE)
 }
 
+# From `?base::Syntax` but lightly organized into groups
 .infixes_assign <- c("<-", "<<-", "->", "->>", "=")
 .infixes_comp <- c("==", "!=", ">", ">=", "<", "<=")
+.infixes_operator <- c(
+  "$", "@", "[", "[[", "^", "-", "+", ":", "*", "/",
+  "!", "&", "&&", "|", "||", "~", "?"
+)
 .infixes <- c(
-  "+", "-", "*", "/", "^", "$", "[", "[[", "!", "%%", "%/%", "%in%",
+  "%%", "%/%", "%in%",
```

Comment:
We could also include three more base R infixes: `%*%`, `%x%`, and `%o%`. 

This is probably beyond the scope of this PR, but perhaps we should detect anything wrapped in `%` to cover infix operators from packages.

Reply:
Good point, I added support for arbitrary infix operators in 1f3e5fb692605d05f48751b10ea32531153f6010

Reply:
That looks great!